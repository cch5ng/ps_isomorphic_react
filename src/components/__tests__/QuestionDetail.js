import React from 'react';
import renderer from 'react-test-renderer';
import { mapStateToProps, QuestionDetailDisplay } from '../QuestionDetail'

describe('question detail component', () => {
  describe('Container element', () => {
    describe('mapStateToProps', () => {

      it('should map state to props correctly', () => {
        const sampleQuestion = {
          question_id: 42,
          body: "blah blah blah"
        }
        const appState = {
          questions: [sampleQuestion]
        }
        const ownProps = {
          question_id: 42
        }
        const componentState = mapStateToProps(appState, ownProps)
        expect(componentState).toEqual(sampleQuestion)
        
      })
    })
  })

  describe('Display element', () => {
    it('should not regress', () => {
      //{title,body,answer_count,tags}

      const tree = renderer.create(
        <QuestionDetailDisplay 
          title="haha!"
          body="blah blah"
          answer_count={3}
          tags={['hey', 'you']}
        />
      )
      expect(tree.toJSON()).toMatchSnapshot()
    })

  })

})