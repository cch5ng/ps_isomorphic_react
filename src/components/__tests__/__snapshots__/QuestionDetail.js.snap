// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question detail component Display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    haha!
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hey
        </code>
        <code>
          you
        </code>
      </div>
    </div>
    <div>
      <p>
        blah blah
      </p>
    </div>
    <div>
      3
       Answers
    </div>
  </div>
</div>
`;
